### ISSUE TITLE
Sharing data is never available in the App [iOS]

### ISSUE THREAD

**Issue by yurayavorski** [achorein/expo-share-intent#133]
**Describe the bug**
After setting up my app following the installation process I can't access any shared data in my app. The app is available in the list of sharing options, but after app is selected it navigates you to the app, but share intent hooks are never re-rendered and no data is available in the app. I tried both approaches with `useShareIntent` and `ShareIntentProvider` but neither works. 

**To Reproduce**
share any kind of data to the app, no data is available in the app

**Environment**
System:
  OS: macOS 14.5
  CPU: (12) arm64 Apple M2 Max
  Memory: 87.31 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.19.1
    path: ~/.nvm/versions/node/v18.19.1/bin/node
  Yarn: Not Found
  npm:
    version: 10.2.4
    path: ~/.nvm/versions/node/v18.19.1/bin/npm
  Watchman:
    version: 2024.07.15.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.5
      - iOS 17.5
      - macOS 14.5
      - tvOS 17.5
      - visionOS 1.2
      - watchOS 10.5
  Android SDK: Not Found
IDEs:
  Android Studio: 2023.3 AI-233.14808.21.2331.11926650
  Xcode:
    version: 15.4/15F31d
    path: /usr/bin/xcodebuild
Languages:
  Java: Not Found
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.74.2
    wanted: 0.74.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: true
  newArchEnabled: false

**Additional context**

- expo version (ex: 49) : 51
- using firebase :  no
- using static build : tried both static build and eas
- routing : tried both static and router-navigation
- package version (ex: 1.2.0) : 2.5.2
- platform target : iOS
- device : real device
- type: tried both local and expo build

**Comment by davram88** [comment]
I am getting exactly the same issue as above, with the same environment as yours. It works good on android expo dev client, but the IOS is not working.

**Comment by achorein** [comment]
can you give me your app.json configuration and from which app you try to share (what type of content ?)

**Comment by achorein** [comment]
sounds lile a problem with Expo Linking (`useLinkingURL`), make sure you use this package in the top of your app.

**Comment by davram88** [comment]
`const IS_DEV = process.env.APP_VARIANT === 'development';
const IS_STAGING = process.env.APP_VARIANT === 'staging';
const IS_PRODUCTION = process.env.APP_VARIANT === 'production';

const SPLASH_CONFIG = {
  backgroundColor: '#ffffff',
  image: './assets/splash.png',
  resizeMode: 'cover',
}
const DARK_SPLASH_CONFIG = {
  backgroundColor: '#001429',
  image: './assets/splash-dark.png',
  resizeMode: 'cover',
}

const SPLASH_CONFIG_ANDROID = {
  backgroundColor: '#0F52BA',
  image: './assets/splash.png',
  resizeMode: 'cover',
}
const DARK_SPLASH_CONFIG_ANDROID = {
  backgroundColor: '#0f141b',
  image: './assets/splash-dark.png',
  resizeMode: 'cover',
}

export default {
  name: 
  slug: 
  owner:
  version: '2.0.2',
  orientation: 'portrait',
  primaryColor: '#ffffff',
  icon: './assets/images/icon.png',
  scheme: 
  userInterfaceStyle: 'automatic',
  jsEngine: 'hermes',
  splash: SPLASH_CONFIG,
  notification: {
    icon: './assets/ic_stat_onesignal_default.png',
    color: '#ffffff'
  },
  assetBundlePatterns: [
    '**/*'
  ],
  ios: {
    userInterfaceStyle: 'automatic',
    supportsTablet: false,
    softwareKeyboardLayoutMode: 'pan',
    bundleIdentifier: 
    googleServicesFile: 
    splash: {
      ...SPLASH_CONFIG,
      dark: DARK_SPLASH_CONFIG,
    },
    config: {
      usesNonExemptEncryption: false
    },
    infoPlist: {
      NSFaceIDUsageDescription: 'Allow to use Face ID.',
      NSCameraUsageDescription:
        'Used for profile pictures, posts, and other kinds of content.',
      NSMicrophoneUsageDescription:
        'Used for posts and other kinds of content.',
      NSPhotoLibraryAddUsageDescription:
        'Used to save images to your library.',
      NSPhotoLibraryUsageDescription:
        'Used for profile pictures, posts, and other kinds of content',
      CFBundleSpokenName: '',
    },
    runtimeVersion: {
      policy: 'appVersion'
    },
    associatedDomains: 
  },
  androidStatusBar: {
    barStyle: 'light-content',
    backgroundColor: '#00000000',
  },
  // Dark nav bar in light mode is better than light nav bar in dark mode
  androidNavigationBar: {
    barStyle: 'light-content',
    backgroundColor: DARK_SPLASH_CONFIG_ANDROID.backgroundColor,
  },
  android: {
    googleServicesFile: IS_DEV ? './google-services-dev.json' : IS_STAGING ? './google-services-staging.json' : './google-services.json',
    package: 
    softwareKeyboardLayoutMode: 'pan',
    userInterfaceStyle: 'automatic',
    splash: {
      ...SPLASH_CONFIG_ANDROID,
      dark: DARK_SPLASH_CONFIG_ANDROID,
    },
    adaptiveIcon: {
      foregroundImage: './assets/images/adaptive-icon.png',
      backgroundColor: '#ffffff'
    },
    permissions: [
      'android.permission.USE_BIOMETRIC',
      'android.permission.USE_FINGERPRINT'
    ],
    intentFilters: [
      {
        action: 'VIEW',
        autoVerify: true,
        data: [],
        category: ['BROWSABLE', 'DEFAULT']
      }
    ]
  },
  plugins: [
    [
      "expo-image-picker",
      {
        "photosPermission": "The app accesses your photos to let you share them with your friends."
      }
    ],
    [
      'onesignal-expo-plugin',
      {
        mode: IS_PRODUCTION ? 'production' : 'development',
        smallIcons: ['./assets/ic_stat_onesignal_default.png'],
        largeIcons: ['./assets/ic_onesignal_large_icon_default.png']
      }
    ],
    [
      'expo-notifications',
      {
        icon: './assets/ic_stat_onesignal_default.png',
        color: '#ffffff'
      }
    ],
    [
      'expo-build-properties',
      {
        ios: {
          useFrameworks: 'static'
        }
      }
    ],
    'expo-localization',
    [
      'expo-local-authentication',
      {
        faceIDPermission: 'Allow to use Face ID.'
      }
    ],
    "@stream-io/video-react-native-sdk",
    [
      "@config-plugins/react-native-webrtc",
      {
        "cameraPermission": "Allow to access your camera",
        "microphonePermission": "Allow to access your microphone"
      }
    ],
    [
      "@sentry/react-native/expo",
      {
        "organization": "",
        "project": "",
        // If you are using a self-hosted instance, update the value of the url property
        // to point towards your self-hosted instance. For example, https://self-hosted.example.com/.
        "url": "https://sentry.io/"
      }
    ],
    [
      'expo-share-intent',
      {
        "iosActivationRules": {
          "NSExtensionActivationSupportsText": true,
          "NSExtensionActivationSupportsWebURLWithMaxCount": 1,
          "NSExtensionActivationSupportsWebPageWithMaxCount": 1,
          "NSExtensionActivationSupportsImageWithMaxCount": 1,
          "NSExtensionActivationSupportsMovieWithMaxCount": 1,
        },
        "androidIntentFilters": ["text/*", "image/*"]
      }
    ],
    './plugins/withAndroidSplashScreenStatusBarTranslucentPlugin.js',
    'expo-font',
    'expo-secure-store',
    'expo-asset'
  ],
  extra: {
    oneSignalAppId:
    eas: {
      projectId: ''
    }
  }
}
`

This is my app. config.js I removed sensitive data.. To your question, I did a test by going to yahoo finance and cnbc on safari and trying to share those articles.

As per your comment 'sounds lile a problem with Expo Linking (useLinkingURL), make sure you use this package in the top of your app.'. Do you mean on top of the App.tsx file?

**Comment by achorein** [comment]
@davram88 yes, using the `ShareIntentProvider` in the `App.tsx` is the safer usage.

your `app.json` looks great, just few things :
- `intentFilters` should not be necessary in android section
- i'm not sure that static build work greats (`useFrameworks: 'static'`)

do you have a log with the correct scheme detected ?
```
 DEBUG  expoShareIntent[scheme] from expoConfig: <expo.scheme>
 ```
